cremalab/crema-app-web
# add redux slice and store templates with prompt options

## Description for [TASK_ID](TASK_URL)

Provide some opinionated [Hygen](https://www.hygen.io/) templates for creating redux state using [Redux Toolkit](https://redux-toolkit.js.org/)

Added the following commands to do so:
  - `npm run new:redux:slice`
  - `npm run new:redux:store`


### 📋 Acceptance Criteria

> Checked off by the PR **Assignee(s)**

- [ ] Create new store with `npm run new:store`
  - [ ] Works with thunk?
  - [ ] File output as expected
  - [ ] Checks if store exists before continuing
- [ ] Create new slice with `npm run new:slice`
  - [ ] File output as expected with dynamic name
  - [ ] Inject new slice/reducer imports into store and root reducers
  - [ ] Outputs example if requested


### 👩‍🔬 Test Instructions

1. Create new store with `npm run new:store`
  - Works with thunk?
  - File output as expected
  - Checks if store exists before continuing
2. Create new slice with `npm run new:slice`
  - File output as expected with dynamic name
  - Inject new slice/reducer imports into store and root reducers
  - Outputs example if requested

## 🔎 Reviewer Checklist

> Checked off by the PR **Reviewers**

### Required

> These always need to be checked


### Additional

> Delete if unneeded

- [ ] Code has been tested and confirmed locally
- [ ] Generated tests for slices should pass.
- [ ] Is the directory name `store` sufficient?  I know others have used names such as `state`.  I'm good with whatever, `store` is just what I've used.

---

>### Roles & Responsibilities
>
>#### 👨‍💻 Assignee
>
>- Initiator of this PR (be sure to set in GitHub UI)
>- Addresses feedback and change requests
>- Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
>#### 👩‍💻 Reviewer
>
>- Invited to review PR by **Assignee** (via GitHub UI)
>- Is expected to complete a review and address followup
